### DISCUSSION TITLE
bug: Code suggestions from Copilot in LazyVim are not indented properly

### DISCUSSION THREAD

**Discussion by yzzting** [LazyVim/LazyVim#1941] [Q&A]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.9.4

### Operating system/version

MacOS 14.1

### Describe the bug

I am encountering an issue with the GitHub Copilot integration within LazyVim. While the code suggestions are functioning as expected and providing relevant results, the inserted code does not maintain correct indentation. 

This is causing me to manually adjust the indentation every time I accept a suggestion, which is not ideal for my workflow. I am not sure if this issue is due to a specific configuration on my end or a bug within the integration itself.

Any guidance or suggestions on how to address this issue would be greatly appreciated. Thanks in advance!

![ezgif-1-93172c0a33](https://github.com/LazyVim/LazyVim/assets/15907064/f95ada55-ece6-4036-aaef-92341b1a376e)

```lua

  {
    "zbirenbaum/copilot.lua",
    enabled = true,
    cmd = "Copilot",
    event = "InsertEnter",
    opts = {
      suggestion = { enabled = false },
      panel = { enabled = false },
    },
  },
  {
    "nvim-cmp",
    dependencies = {
      "hrsh7th/cmp-emoji",
      {
        "zbirenbaum/copilot-cmp",
        opts = {},
      },
    },
    ---@param opts cmp.ConfigSchema
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.sources = cmp.config.sources(vim.list_extend(opts.sources, { { name = "copilot" }, { name = "emoji" } }))
    end,
  },

```

### Steps To Reproduce

1. **Code Suggestions** - When I am coding in LazyVim, I get code suggestions from GitHub's Copilot as expected.

2. **Selecting a Suggestion** - I select one of the suggested code snippets provided by Copilot to insert into my code.

3. **Indentation Issue** - The inserted code snippet does not maintain the correct indentation. This happens every time I accept a suggestion, forcing me to manually adjust the indentation, which disrupts my workflow.

### Expected Behavior

Insert at cursor position

**Comment by folke** [comment]
Please use discussions for this kind of questions.

**Comment by yzzting** [comment]
OK，So how do I deal with this？

**Reply by ryansch** [reply]
I've been having this issue with ruby for quite some time.

**Reply by dpetka2001** [reply]
Won't opening an issue at the appropriate plugin's repo yield hopefully better results? There's nothing that can be done here if a plugin isn't behaving as it should.
